### PR TITLE
:sparkles: Rebase for k/k 1.32

### PR DESCRIPTION
**Pull request by gman0**
## Related issue(s)

Part of https://github.com/kcp-dev/kcp/issues/3209

### CODE REVIEW

**Comment by gman0** [comment]
I've marked this PR as draft for the moment until I find out why golangci-lint is crashing:

```
ERRO [linters_context/goanalysis] nilness: panic during analysis: internal error: unhandled type *ir.ArrayConst, goroutine 4064 [running]:
runtime/debug.Stack()
	/usr/lib/golang/src/runtime/debug/stack.go:26 +0x5e
github.com/golangci/golangci-lint/pkg/goanalysis.(*action).analyzeSafe.func1()
	/home/rvasek/go/pkg/mod/github.com/golangci/golangci-lint@v1.58.1/pkg/goanalysis/runner_action.go:105 +0x5a
panic({0x17a6b20?, 0xc00414c0e0?})
	/usr/lib/golang/src/runtime/panic.go:785 +0x132
honnef.co/go/tools/analysis/facts/nilness.impl.func1({0x7fdb84141650, 0xc00553b860})
	/home/rvasek/go/pkg/mod/honnef.co/go/tools@v0.4.7/analysis/facts/nilness/nilness.go:239 +0x8a5
honnef.co/go/tools/analysis/facts/nilness.impl.func1({0x1cbec38, 0xc0013a6c00})
	/home/rvasek/go/pkg/mod/honnef.co/go/tools@v0.4.7/analysis/facts/nilness/nilness.go:147 +0x90e
honnef.co/go/tools/analysis/facts/nilness.impl(0xc0012c10a0, 0xc000fa57c0, 0xc0055abcb8)
	/home/rvasek/go/pkg/mod/honnef.co/go/tools@v0.4.7/analysis/facts/nilness/nilness.go:246 +0x32a
honnef.co/go/tools/analysis/facts/nilness.run(0xc0012c10a0)
	/home/rvasek/go/pkg/mod/honnef.co/go/tools@v0.4.7/analysis/facts/nilness/nilness.go:66 +0x125
github.com/golangci/golangci-lint/pkg/goanalysis.(*action).analyze(0xc0034a35a0)
	/home/rvasek/go/pkg/mod/github.com/golangci/golangci-lint@v1.58.1/pkg/goanalysis/runner_action.go:191 +0x9cd
github.com/golangci/golangci-lint/pkg/goanalysis.(*action).analyzeSafe.func2()
	/home/rvasek/go/pkg/mod/github.com/golangci/golangci-lint@v1.58.1/pkg/goanalysis/runner_action.go:113 +0x17
github.com/golangci/golangci-lint/pkg/timeutils.(*Stopwatch).TrackStage(0xc000ec7040, {0x19cb761, 0x7}, 0xc003eec748)
	/home/rvasek/go/pkg/mod/github.com/golangci/golangci-lint@v1.58.1/pkg/timeutils/stopwatch.go:111 +0x44
github.com/golangci/golangci-lint/pkg/goanalysis.(*action).analyzeSafe(0xc00150d500?)
	/home/rvasek/go/pkg/mod/github.com/golangci/golangci-lint@v1.58.1/pkg/goanalysis/runner_action.go:112 +0x6e
github.com/golangci/golangci-lint/pkg/goanalysis.(*loadingPackage).analyze.func2(0xc0034a35a0)
	/home/rvasek/go/pkg/mod/github.com/golangci/golangci-lint@v1.58.1/pkg/goanalysis/runner_loadingpackage.go:80 +0xa5
created by github.com/golangci/golangci-lint/pkg/goanalysis.(*loadingPackage).analyze in goroutine 1114
	/home/rvasek/go/pkg/mod/github.com/golangci/golangci-lint@v1.58.1/pkg/goanalysis/runner_loadingpackage.go:75 +0x1e9
```

File: Makefile [hunk 1]
```diff
@@ -19,7 +19,7 @@ GO_INSTALL = ./hack/go-install.sh
 TOOLS_DIR=hack/tools
 GOBIN_DIR := $(abspath $(TOOLS_DIR))
 
-GOLANGCI_LINT_VER := v1.58.1
+GOLANGCI_LINT_VER := v1.62.2
```
Comment (mjudeikis) [inline Makefile:22]: If we revert, still fails linting?

Comment (gman0) [reply Makefile:22]: It crashes immediately after running `make lint test` locally, even without any changes done. The stack trace suggests the issue is inside golangci-lint itself. Indeed it's strange that it should crash when it must have worked before. In any case, do we want to keep the older version at all? Or we update in a separate PR?

**Comment by gman0** [comment]
```
docker run --rm ghcr.io/kcp-dev/infra/build:1.22.10-1 go version
go version go1.22.10 linux/amd64
```

1.22.10-1 is the latest tag we have. Can we build a new one with more recent go toolkit?

**Pull request merged**